### ISSUE TITLE
SC2010 should not warn against a negated matcher (grep -v)

### ISSUE THREAD

**Issue by ecbrodie** [koalaman/shellcheck#1086]
#### For bugs
- Rule Id: SC2010
- My shellcheck version: 0.4.7
- [ ] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
ls | grep -v IGNORE_ME | while read -r line; do
  cd "$line" || exit
  bundle
  cd ..
done
```

#### Here's what shellcheck currently says:

```
^-- SC2010: Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.
```

#### Here's what I wanted or expected to see:

This warning should not appear. [According to the SC2010 wiki page](https://github.com/koalaman/shellcheck/wiki/SC2010), the exception to this rule is a negated matcher via `grep -v`, which is indeed what my code does.

**Comment by Nightfirecat** [comment]
Assuming extglob is set, the loop condition could be more simply written as `for line in !(*IGNORE_ME*); do`. May help in your case but given that dependency (and given that globs don't match grep's full regex capabilities), I don't know that this is a suitable replacement in the general case.

**Comment by koalaman** [comment]
The wiki page was wrong, I've fixed it. 

In this case (bash), an extglob like @Nightfirecat suggests would be the way to go.

**Comment by ecbrodie** [comment]
I'm happy to see this issue got the wiki page to be corrected :)
